index.html

```diff
@@ -0,0 +1,114 @@
+<!DOCTYPE html>
+<html>
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Roman Numeral Converter</title>
+  <style>
+    body {
+      font-family: Arial, sans-serif;
+      text-align: center;
+      background: linear-gradient(90deg, white, green, black);
+      color: white;
+      margin: 0;
+      padding: 20px;
+      box-sizing: border-box;
+    }
+
+    h1 {
+      color: black;
+    }
+
+    label {
+      display: block;
+      font-size: 20px;
+      margin-bottom: 10px;
+    }
+
+    input[type="number"] {
+      width: 200px;
+      height: 30px;
+      font-size: 18px;
+      padding: 5px;
+    }
+
+    button {
+      background-color: green;
+      color: white;
+      font-size: 20px;
+      padding: 10px 20px;
+      border: none;
+      cursor: pointer;
+    }
+
+    button:hover {
+      background-color: darkgreen;
+    }
+
+    #output {
+      font-size: 24px;
+      margin-top: 20px;
+    }
+  </style>
+</head>
+<body>
+  <h1>Roman Numeral Converter</h1>
+  <label for="number">Enter a Number:</label>
+  <input type="number" id="number">
+  <br><br>
+  <button id="convert-btn">Convert</button>
+  <br><br>
+  <div id="output"></div>
+
+  <script>
+    const convertButton = document.getElementById('convert-btn');
+    const outputDiv = document.getElementById('output');
+
+    convertButton.addEventListener('click', function() {
+      const numberInput = document.getElementById('number').value;
+
+      if (numberInput.trim().length === 0) {
+        outputDiv.innerText = 'Please enter a valid number';
+      } else {
+        const number = parseInt(numberInput);
+
+        if (isNaN(number) || number < 1) {
+          outputDiv.innerText = 'Please enter a number greater than or equal to 1';
+        } else if (number >= 4000) {
+          outputDiv.innerText = 'Please enter a number less than or equal to 3999';
+        } else {
+          outputDiv.innerText = convertToRoman(number);
+        }
+      }
+    });
+
+    function convertToRoman(number) {
+      const numeralMap = [
+        { value: 1000, numeral: 'M' },
+        { value: 900, numeral: 'CM' },
+        { value: 500, numeral: 'D' },
+        { value: 400, numeral: 'CD' },
+        { value: 100, numeral: 'C' },
+        { value: 90, numeral: 'XC' },
+        { value: 50, numeral: 'L' },
+        { value: 40, numeral: 'XL' },
+        { value: 10, numeral: 'X' },
+        { value: 9, numeral: 'IX' },
+        { value: 5, numeral: 'V' },
+        { value: 4, numeral: 'IV' },
+        { value: 1, numeral: 'I' }
+      ];
+
+      let roman = '';
+      for (let i = 0; i < numeralMap.length; i++) {
+        while (number >= numeralMap[i].value) {
+          roman += numeralMap[i].numeral;
+          number -= numeralMap[i].value;
+        }
+      }
+
+      return roman;
+    }
+  </script>
```

Comment:
- You need to have your JS code in a separate file like `script.js` and add its link using `<script>` tag at the end of your <body>` in your HTML file.